iroh/src/magicsock.rs

```diff
@@ -979,6 +999,23 @@ impl Handle {
     #[instrument(skip_all)]
     pub(crate) async fn close(&self) {
         trace!(me = ?self.public_key, "magicsock closing...");
+        if self.msock.is_closed() {
+            return;
+        }
+
+        // Only set closing if not already set
+        if self
+            .msock
+            .closing
+            .compare_exchange(false, true, Ordering::Acquire, Ordering::Relaxed)
+            .is_err()
+        {
+            return;
```

Comment:
This means that if you call `Endpoint::close` twice on a clone of the endpoint, only the first call will wait for `quinn::Endpoint::wait_idle`, whereas the second call will complete immediately. I think we should always await `Endpoint::wait_idle`, even if  `closing` is already true.

Reply:
fixed, though we never actually come here, because `Endpoint::close` shortcircuits way before this